lib/route-recognizer.js

```diff
@@ -497,7 +497,7 @@ RouteRecognizer.prototype = {
       queryParams = this.parseQueryString(queryString);
     }
 
-    path = decodeURI(path);
+    path = decodeURIComponent(path);
```

Comment:
@mmun this is the fix, but you leave the decodeURI, decodeURI only decodes characters that are not required for URI structure, it will not touch structural escapes, so it can be done before parsing.